odoo/osv/expression.py

```diff
@@ -755,6 +755,9 @@ def child_of_domain(left, ids, left_model, parent=None, prefix=''):
                 (when available), or as an expanded [(left,in,child_ids)] """
             if not ids:
                 return [FALSE_LEAF]
+            if not any(ids):
```

Comment:
```suggestion
            if not all(ids):
```

?

Reply:
Indeed, we can say that domains
```python
[('xxx', 'child_of', [False] + more_ids)]
```
and
```python
['|', ('xxx', 'child_of', [False]), ('xxx', 'child_of', more_ids)]
```
are *logically* equivalent, and as the left-hand side of the OR is equivalent to true, the whole stuff is equivalent to true.  So you can use the condition `if not all(ids)`...